migrations/201906170001-Transaction-fixTaxAmount.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+import models, { sequelize, Op } from '../server/models';
+import Promise from 'bluebird';
+
+const fetchTransactions = async () => {
+  const query = { where: { taxAmount: { [Op.gt]: 0 } } };
+  if (process.env.LIMIT) {
+    query.limit = process.env.LIMIT;
+  }
+  return await models.Transaction.findAll(query);
+};
+
+const toNegative = n => (n > 0 ? -n : n);
+
+const fixTransaction = transaction => {
+  const updatedTransaction = { ...transaction.dataValues };
+  delete updatedTransaction.id;
+  updatedTransaction.taxAmount = toNegative(transaction.taxAmount);
+  if (transaction.type === 'CREDIT') {
+    updatedTransaction.netAmountInCollectiveCurrency =
+      transaction.netAmountInCollectiveCurrency + updatedTransaction.taxAmount;
+  } else if (transaction.type === 'DEBIT') {
+    updatedTransaction.amount = transaction.amount - updatedTransaction.taxAmount;
+    updatedTransaction.amountInHostCurrency = transaction.amountInHostCurrency - updatedTransaction.taxAmount;
+  }
+  return updatedTransaction;
+};
+
+const processTransaction = async transaction => {
+  let dbTransaction;
+  const updatedTransaction = fixTransaction(transaction);
+  console.log('*******************************************************');
+  console.log('>>> transaction.description', transaction.dataValues.description);
+  console.log('>>> transaction.TransactionGroup', transaction.dataValues.TransactionGroup);
+  console.log('*******************************************************');
+  console.log('>>> transaction.type', transaction.dataValues.type);
+  console.log('>>> transaction.amount', transaction.dataValues.amount);
+  console.log('>>> transaction.amountInHostCurrency', transaction.dataValues.amountInHostCurrency);
+  console.log('>>> transaction.taxAmount', transaction.dataValues.taxAmount);
+  console.log('>>> transaction.netAmountInCollectiveCurrency', transaction.dataValues.netAmountInCollectiveCurrency);
+  console.log('*******************************************************');
+  console.log('>>> updatedTransaction.type', updatedTransaction.type);
+  console.log('>>> updatedTransaction.amount', updatedTransaction.amount);
+  console.log('>>> updatedTransaction.amountInHostCurrency', updatedTransaction.amountInHostCurrency);
+  console.log('>>> updatedTransaction.taxAmount', updatedTransaction.taxAmount);
+  console.log('>>> updatedTransaction.netAmountInCollectiveCurrency', updatedTransaction.netAmountInCollectiveCurrency);
+  console.log('*******************************************************\n\n');
+  try {
+    dbTransaction = await sequelize.transaction();
+    await transaction.destroy({ transaction: dbTransaction });
```

Comment:
Golden rule in any ledger: everything should be immutable. We shouldn't ever edit a transaction. The way to do it is to mark it as deleted and create a new one so that we keep the history.

Where do we reference a transaction using its id? (we shouldn't)

Reply:
I was thinking about `RefundTransactionId`

Reply:
Ah ok, well I'd argue that it's good to see the state of the transaction at the time it was refunded. So it's good.

I've added the list of attributes to fetch.
Anything else? 
Can you approve?